src/libos/src/time/timer_file.rs

```diff
@@ -0,0 +1,385 @@
+        } else {
+            inner.status = TimerFdStatus::LOOP;
+        }
+
+        inner.alarm_cnt = 0;
+        inner.last_alarm_cnt = 0;
+
+        // Start background poll task to monitor timerfd events
+        let join_handle = SpawnOptions::new({
+            let pollee = self.pollee.clone();
+            let mut it_value = match inner.fire_time.checked_sub(cur_time) {
+                Some(duration) => duration,
+                _ => Duration::new(0, 0),
+            };
+
+            let interval = inner.cur_interval;
+
+            async move {
+                let waiter = Waiter::new();
+                if let Ok(_) = waiter.wait_timeout(Some(&mut it_value)).await {
+                    // TODO: support interrupt sleep.
+                    // return_errno!(EINTR, "sleep interrupted");
+                    unreachable!("this waiter can not be interrupted");
+                }
+
+                trace!("timerfd trigger it_value");
+                pollee.add_events(Events::IN);
+
+                if !interval.is_zero() {
+                    loop {
+                        let mut timeout = interval;
+                        let waiter = Waiter::new();
+                        let res = waiter.wait_timeout(Some(&mut timeout)).await;
+                        match res {
+                            Err(e) => {
+                                if e.errno() == EINTR {
+                                    break;
+                                } else {
+                                    trace!("timerfd trigger it_interval {:?}", interval);
+                                    pollee.add_events(Events::IN);
+                                }
+                            }
+                            _ => {
+                                panic!("impossible as there is no waker or timeout");
+                            }
+                        }
+                    }
+                }
+
+                trace!("Timerfd poll task end");
+            }
+        })
+        .priority(async_rt::sched::SchedPriority::Low)
+        .spawn();
+
+        inner.task_handle = Some(join_handle);
+
+        Ok(TimerfileDurations {
+            it_interval: old_it_interval,
+            it_value: old_it_value,
+        })
+    }
+
+    pub fn time(&self) -> Result<TimerfileDurations> {
+        let mut ret_time = TimerfileDurations::default();
+        let mut inner = self.inner.lock().unwrap();
+
+        let interval = inner.cur_interval;
+        let cur_time =
+            timespec_t::from(vdso_time::clock_gettime(self.clockid).unwrap()).as_duration();
+
+        if inner.status != TimerFdStatus::STOP {
+            let delta = match inner.fire_time.checked_sub(cur_time) {
+                Some(duration) => duration,
+                _ => Duration::new(0, 0),
+            };
+
+            if inner.status == TimerFdStatus::ONE_SHOT {
+                if delta.is_zero() == true {
+                    inner.alarm_cnt = 1;
```

Comment:
Do not get the point why it should be 1? I think it could be 0 too.

Reply:
`alarm_cnt=1` means there is one expired timer because the delta ( the delta of expected fire time and current time) is 0.
Anyway, I will try re-design the whole status/timer check to make it more readble.